authors/apps/comments/views.py

```diff
@@ -0,0 +1,211 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework.generics import GenericAPIView
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework import permissions
+from rest_framework.renderers import JSONRenderer
+
+from .serializers import CommentSerializer, CommentReplySerializer
+from .models import Comment, CommentReply
+from ..profiles.models import Profile
+from ..articles.models import Article
+from ..articles.utils import model_helpers
+
+
+class CommentApiView (GenericAPIView):
+    """
+    The ApiView class handles the addition of a comment to an article
+    and retrieving of all comments to an article.
+    """
+    permission_classes = (permissions.IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentSerializer
+    renderer_classes = (JSONRenderer,)
+
+    def get_required_objects(self, request, slug):
+        self.article = get_object_or_404(Article, slug=slug)
+        self.data = request.data.get(
+            'comment') if 'comment' in request.data else request.data
+
+    def get_author_profile(self, request):
+        self.author = Profile.objects.get(user=request.user)
+
+    def get(self, request, slug):
+        """Retrieve all comments on an article"""
+        self.get_required_objects(request, slug)
+        comments = Comment.objects.filter(article=self.article)
```

Comment:
Don't you think it would make sense to sort the comments by `timestamp`?

Reply:
Yes, you are right @Karuhanga . The ordering on the serializer depends on the `created_at` field